arch/arc/mm/tlb-arcv3.c

```diff
@@ -60,6 +61,91 @@ int arc_mmu_mumbojumbo(int c, char *buf, int len)
 	return n;
 }
 
+/*
+ * At this point we mapped kernel code (PAGE_OFFSET to _end) in head.S.
+ * Assumes we have PGD and PUD
+ * TODO: assumes 4 levels, implement properly using p*d_addr_end loops
+ */
+static pmd_t fixmap_pmd[PTRS_PER_PMD] __page_aligned_bss;
+static pte_t fixmap_pte[PTRS_PER_PTE] __page_aligned_bss;
+void early_fixmap_init(void)
+{
+	unsigned long addr;
+	pgd_t *pgd;
+	p4d_t *p4d;
+	pud_t *pud;
+	pmd_t *pmd;
+
+	/* Make sure fixmap fits in one PMD. */
+	BUILD_BUG_ON(pmd_index(FIXADDR_START) != \
+		     pmd_index(FIXADDR_START + FIXADDR_SIZE));
+
+	addr = FIXADDR_START;
+
+	pgd = (pgd_t *) __va(read_aux_64(ARC_REG_MMU_RTP1));
```

Comment:
Why don't you just use `early_pg_dir`  here ?

Reply:
Just not to care about it if we will change something in future. Or if this function will be called from some other place.